plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -86,62 +92,77 @@ function isSupportedTemplate(entity: TemplateEntityV1beta3) {
  * until someone explicitly passes an IdentityApi. When we have reasonable confidence that most backstage deployments
  * are using the IdentityApi, we can remove this function.
  */
-function buildDefaultIdentityClient({
-  logger,
-}: {
-  logger: Logger;
-}): IdentityApi {
-  return {
-    getIdentity: async ({ request }: IdentityApiGetIdentityRequest) => {
-      const header = request.headers.authorization;
 
-      if (!header) {
-        return undefined;
+class CustomIdentityClient implements IdentityApi {
```

Comment:
I needed to convert this custom implementation of the IdentityApi to a class to be able to properly implement the overloaded function for it.